source/common/config/well_known_names.cc

```diff
@@ -24,97 +24,95 @@ TagNameValues::TagNameValues() {
   // - Typical * notation will be used to denote an arbitrary set of characters.
 
   // *_rq(_<response_code>)
-  addRegex(RESPONSE_CODE, "_rq(_(\\d{3}))$", "_rq_");
+  addRe2(RESPONSE_CODE, R"(_rq(_(\d{3}))$)", "_rq_");
 
   // *_rq_(<response_code_class>)xx
-  addRegex(RESPONSE_CODE_CLASS, "_rq_(\\d)xx$", "_rq_");
+  addRe2(RESPONSE_CODE_CLASS, R"(_rq_((\d))xx$)", "_rq_");
 
   // http.[<stat_prefix>.]dynamodb.table.[<table_name>.]capacity.[<operation_name>.](__partition_id=<last_seven_characters_from_partition_id>)
-  addRegex(DYNAMO_PARTITION_ID,
-           "^http(?=\\.).*?\\.dynamodb\\.table(?=\\.).*?\\."
-           "capacity(?=\\.).*?(\\.__partition_id=(\\w{7}))$",
-           ".dynamodb.table.");
+  addRe2(DYNAMO_PARTITION_ID,
+         R"(^http\.[\w-]+\.dynamodb\.table\.[\w-]+\.capacity\.[\w-]+(\.__partition_id=(\w{7}))$)",
```

Comment:
one thought on how these segments are expressed in code: I wonder if we should define some infrastructure like macros or helper functions to provide some of these symbolically rather than spelling out the full regexes. That might make them easier to read for non-regex-experts, and also have a single point where we define what a "word" means, e.g. (alphanumerics plus -) vs (everything except .)

Reply:
how about the `expandRegex()` function I added?